spec/solver_spec.rb

```diff
@@ -0,0 +1,21 @@
+require_relative 'spec_helper'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  context 'To test the solver class' do
+    it 'Returns the factorial of the number' do
+      expect(@solver.factorial(5)).to eq 120
+    end
```

Comment:
- [ ] Kindly add a test case that handles negative values which raise an exception. Also, consider adding a test case when input is 0, the result should equal 1.

Reply:
I have added a test that raises an argument error when the number is negative

Reply:
@omarsalem7 , i added the test for a negative number, but when the test runs it raises an argument error, which is correct as per my code. However the test fails because its an error raised. Please refer to the screenshots
![image](https://user-images.githubusercontent.com/81410040/190408987-1215aff6-9313-4240-8b5b-b37cfed1da95.png)
![image](https://user-images.githubusercontent.com/81410040/190409092-1c6c1fae-c94b-4b3c-8382-d794c6c04cfa.png)